src/mapping_navigation/scripts/navigation.py

```diff
@@ -2,33 +2,52 @@
 
 import rospy
 from std_msgs.msg import Float64, String
-from mapping_navigation.msg import PathData
 from geometry_msgs.msg import PoseStamped
+from pure_pursuit import *
+from std_msgs.msg import Float64
+from simple_path import *
 
 
 class Navigation:
 
     """
-    This does mapping
+    This does navigation
     """
 
-    def __init__(self):
+    def __init__(self, path: List[Point], look_ahead_distance: float, look_forward_gain: float):
         self.steering_pub = rospy.Publisher('steering', Float64, queue_size=10)
-        self.curve_pub = rospy.Publisher('curve_data', String, queue_size=10)
-        self.path_data = rospy.Publisher('pathData', PathData, queue_size=10)
+        self.path = path
+        self.target_ind = 0
+        self.navigator = PurePursuit(look_ahead_distance, look_forward_gain, path)
+        self.state = State()
 
     def listener(self):
         rospy.init_node('talker', anonymous=True)
         rospy.Subscriber('airsimPose', PoseStamped, self.handle_gps_data)
-        rospy.spin()
+        rospy.Subscriber("sensor/speed", Float64, self.handle_speed_data)        
+        rate = rospy.Rate(30)
 
-    def handle_gps_data(self, data):
-        self.steering_pub.publish(1)
-        self.curve_pub.publish('curve_data')
-        self.path_data.publish(PathData())
-        print('Handles gps data')
+        while not rospy.is_shutdown():
+            self.target_ind, _ = pure_pursuit.search_target_index(state)
+            if len(path)-1 > self.target_ind:
+                steering_angle = self.get_steering_angle()
+                self.steering_pub(steering_pub)
 
+    def handle_gps_data(self, postition: PoseStamped):
+        curr_point = Point((curr_point.pose.position.x, curr_point.pose.position.y))
+        quaterion = (curr_point.pose.orientation.x, curr_point.pose.orientation.y,
+                     curr_point.pose.orientation.z, curr_point.pose.orientation.w)
+        self.state.update_pos(curr_point, quaterion)
+
+    def handle_speed_data(self, speed: Float64):
+        self.state.update_speed(speed.data)
+
+    def get_steering_angle(self) -> float:
```

Comment:
The things is its not really degrees, its airsims steering input which is between -1 and 1, I take it out of degrees when I divide it by 45 degrees, which again was an assumption that 1 was around that angle for the wheels. So I don't think renaming it will make things clearer, its more of a steering ratio value kinda?

Reply:
will do